### PR TITLE
feat(article): add finite state machines article

### PR DESCRIPTION
**Pull request by DaveBitter**

### CODE REVIEW

File: data/blog/finite-state-machines-in-javascript.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+theme: 'blue'
+---
+
+Finite State Machines (FSMs) are a concept in programming that helps model complex systems with a fixed set of states and transitions between those states. In JavaScript, FSMs offer a structured approach to managing application logic for making code more organised, easier to maintain and more.
+
+## What is a Finite State Machine?
+
+An FSM is a model that consists of three main components:
+
+- States: the distinct stages or conditions an application can be in at any given moment
```
Comment (sanderdejong88) [inline data/blog/finite-state-machines-in-javascript.md]: Underline/Bold the first word for each bullet?

File: data/blog/finite-state-machines-in-javascript.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+title: '_Finite State Machines_ in JavaScript'
+date: '2023-07-26'
+tags: ['frontend', 'javascript']
+images: ['/articles/finite-state-machines-in-javascript/finite-state-machines-in-javascript.jpg']
+summary: 'Application state can make any application complex real quick. Let’s have a look at Finite State Machines in Javascript to resolve some of these complexities.'
+authors: ['dave-bitter']
+theme: 'blue'
+---
+
+Finite State Machines (FSMs) are a concept in programming that helps model complex systems with a fixed set of states and transitions between those states. In JavaScript, FSMs offer a structured approach to managing application logic for making code more organised, easier to maintain and more.
```
Comment (sanderdejong88) [inline data/blog/finite-state-machines-in-javascript.md:11]: If I google FSMs I get 0 results which are relevant. So I guess "FSM's" might be better

File: data/blog/finite-state-machines-in-javascript.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    orderStatus = 'Delivered'
+  } else {
+    console.log('Invalid state transition or event.')
+  }
+}
+```
+
+Usually this is the moment I get a coffee to start figuring out what the business logic is in this function. As the codebase grows, more state transitions are added, leading to a convoluted and error-prone `updateOrderStatus` function. The lack of structure and clear separation between states and transitions make it challenging to maintain, debug, and extend the code.
+
+By using a Finite State Machine, we can significantly improve the code's clarity and maintainability. Let's see how the same order status functionality can be implemented with an FSM:
```
Comment (sanderdejong88) [inline data/blog/finite-state-machines-in-javascript.md]: Suddenly no longer using the abbreviation?

File: data/blog/finite-state-machines-in-javascript.md (hunk 2 above)
Comment (DaveBitter) [reply data/blog/finite-state-machines-in-javascript.md:11]: I see, but this is the correct way in english. Probably you get the hits as it is taking FSM and 's separate

File: data/blog/finite-state-machines-in-javascript.md (hunk 3 above)
Comment (DaveBitter) [reply data/blog/finite-state-machines-in-javascript.md]: Updated

File: data/blog/finite-state-machines-in-javascript.md (hunk 1 above)
Comment (DaveBitter) [reply data/blog/finite-state-machines-in-javascript.md]: Updated

**Pull request merged**